source/com.microsoft.tfs.client.clc/src/com/microsoft/tfs/client/clc/commands/Command.java

```diff
@@ -914,6 +915,58 @@ protected final WorkspaceInfo determineCachedWorkspace(final String[] pathFreeAr
         return determineCachedWorkspace(pathFreeArguments, false);
     }
 
+    protected final WorkspaceInfo determineCachedWorkspace(
+        final String[] pathFreeArguments,
+        final boolean ignoreWorkspaceOptionValue)
+        throws CannotFindWorkspaceException,
+            InvalidOptionValueException,
+            InvalidOptionException {
+
+        /*
+         * On all platforms but Mac OSX, we do not have issues with the computer
+         * names detected by CLC.
+         */
+        if (!Platform.isCurrentPlatform(Platform.MAC_OS_X)) {
+            return determineCachedWorkspaceImpl(pathFreeArguments, ignoreWorkspaceOptionValue);
+        }
+
+        /*
+         * On the Mac OSX we should check two versions of the short computer
+         * name to ensure compatibility with the JetBrains' method of its
+         * compilation. The code should be removed when and if this
+         * compatibility is not needed anymore.
+         */
+        WorkspaceInfo workspaceInfo = null;
+        try {
+            workspaceInfo = determineCachedWorkspaceImpl(pathFreeArguments, ignoreWorkspaceOptionValue);
+
+            if (LocalHost.getShortName().equalsIgnoreCase(workspaceInfo.getComputer())) {
+                /*
+                 * The workspace has been created with CLC, we can just use it.
+                 */
+                return workspaceInfo;
+            }
+
+            log.info(
+                MessageFormat.format(
+                    "The computer name {0} of the cached workspace {1} does not match to the current machine {2}", //$NON-NLS-1$
+                    workspaceInfo.getComputer(),
+                    workspaceInfo.getDisplayName(),
+                    LocalHost.getShortName()));
+
+        } catch (final CannotFindWorkspaceException e) {
+            log.info(e.getMessage(), e);
+        }
+
+        final String jbComputerName = LocalHost.getShortNameJB();
+        log.info(MessageFormat.format(
+            "We'll try to find the workspace one more time using the JetBrains name {0}", //$NON-NLS-1$
+            jbComputerName));
+
+        System.setProperty(LocalHost.SHORT_NAME_OVERRIDE_PROPERTY, jbComputerName);
+        return determineCachedWorkspaceImpl(pathFreeArguments, ignoreWorkspaceOptionValue);
```

Comment:
If this final attempt using the JetBrains computer name is unsuccessful, should we set the system property back to its original value?